src/table/settings/game_settings.ini

```diff
@@ -134,6 +137,98 @@ str      = STR_CONFIG_SETTING_DISTANT_JOIN_STATIONS
 strhelp  = STR_CONFIG_SETTING_DISTANT_JOIN_STATIONS_HELPTEXT
 post_cb  = [](auto) { CloseWindowById(WC_SELECT_STATION, 0); }
 
+[SDT_VAR]
+var      = depot.rail_depot_types
+type     = SLE_UINT8
+from     = SLV_EXTENDED_DEPOTS
+flags    = SF_GUI_DROPDOWN
+def      = 1
+min      = 1
+max      = 3
+interval = 1
+str      = STR_CONFIG_SETTING_RAIL_DEPOT_TYPES
+strhelp  = STR_CONFIG_SETTING_RAIL_DEPOT_TYPES_HELPTEXT
+strval   = STR_CONFIG_SETTING_ONLY_STANDARD_DEPOT
+post_cb  = DepotSettingsChanged
+cat      = SC_EXPERT
```

Comment:
I'm wondering if the standard/extended depot determination can be made automatically, instead of by having separate depot types. Essentially having only extended depots which connect to the proper track when connected, and having an option for infinite depot capacity.

But if this is the best way, it should not be hidden in Expert settings. I would argue for Basic, because I suspect these depots will be rather popular (and have seen how players react when important settings are hidden in Expert/Advanced settings...see the uproar over #8688 😕 ).

Reply:
Standard and extended depots cannot be merged into one single type of depots. Standard ones have only one entry/exit side and extended ones have two entry/exit sides. Converting all legacy depots to extended ones is not possible as it would break savegames (allowing standard depots to have two entry/exit sides could join two different train networks).

As the debate about Expert/Basic settings goes on on other threads, I prefer not to discuss it for extended depots. They can be changed later to "Basic" if needed. For me, Expert is ok by now, as players need to know the difference between standard and extended depots and how to use them before changing the setting.